arch/ARM/Nordic/drivers/nrf_common/nrf-twi.adb

```diff
@@ -266,34 +271,29 @@ package body nRF.TWI is
    is
    begin
 
-      This.Do_Stop_Sequence := False;
+      This.Do_Stop_Sequence := True;
 
       case Mem_Addr_Size is
          when Memory_Size_8b =>
             This.Master_Transmit (Addr    => Addr,
-                                  Data    => (0 => UInt8 (Mem_Addr)),
+                                  Data    =>
+                                    (0 => UInt8 (Mem_Addr)) & Data,
```

Comment:
One issue here is that if you send a big `Data` array (e.g. a frame buffer for a screen), you have a pretty high risk of  overflowing the stack with this concatenation operation.

Reply:
True, but (a) is that likely to happen with I2C devices? (b) how could it be addressed, anyway? (c) could we rely on the device being OK with doing the transfer in multiple chunks? (d) what would the chunk size limit be?

I’d imagine that frame buffers would be sent in parts anyway.

We could offer a different interface for chunkable transfers, but you’d still need to concatenate the address and the data given the way TWI works.

Perhaps SUSPEND might offer a solution?

Reply:
OK, I was wrong about this. The second part of my changes was probably a remnant of experimental changes to try to get _some_ handle on what was/wasn’t working; reverted.